docs/content/concepts/resources.mdx

```diff
@@ -5,357 +5,355 @@ description: Resources enable you to separate graph logic from environment, and
+| <PyObject object="ResourceParam"/>               | An annotation used to specify that a plain Python object parameter for an op or asset is a resource.                                                                                                                                    |
+| <PyObject object="ResourceDefinition" />         | Class for resource definitions. You almost never want to use initialize this class directly. Instead, you should extend the <PyObject object="ConfigurableResource" /> class which implements <PyObject object="ResourceDefinition" />. |
+| <PyObject object="InitResourceContext"/>         | The context object provided to a resource during initialization. This object contains required resource, config, and other run information.                                                                                             |
+| <PyObject object="build_init_resource_context"/> | Function for building an <PyObject object="InitResourceContext"/> outside of execution, intended to be used when testing a resource.                                                                                                    |
+| <PyObject object="build_resources"/>             | Function for initializing a set of resources outside of the context of a job's execution.                                                                                                                                               |
+| <PyObject object="with_resources"/>              | Function for providing resources to software-defined assets and source assets.                                                                                                                                                          |
 
 ---
 
 ## Defining a resource
 
-To define a resource, use the <PyObject object="resource" decorator/> decorator. Wrap a function that takes an `init_context` as the first parameter, which is an instance of <PyObject object="InitResourceContext"/>. From this function, return or yield the object that you would like to be available as a resource.
-
-```python file=/concepts/resources/resources.py startafter=start_resource_example endbefore=end_resource_example
-from dagster import resource
-
-
-class ExternalCerealFetcher:
-    def fetch_new_cereals(self, start_ts, end_ts):
-        pass
-
+Typically, resources are defined by subclassing <PyObject object="ConfigurableResource"/>. Attributes on the class are used to define the resource's [configuration schema](/concepts/configuration/config-schema). The configuration system has a few advantages over plain Python parameter passing; configured values are displayed in the Dagster UI and can be set dynamically using environment variables.
```

Comment:
Another advantage is that one can delay the binding of values to configurable at launch-time.